src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Projects/ProjectSystemCache.cs

```diff
@@ -22,7 +22,7 @@ internal class ProjectSystemCache : IProjectSystemCache, IDisposable
         // 0 - Cache is clean
         // 1 - Cache is dirty
         private int _isCacheDirty = 0;
-        private readonly Dictionary<string, CacheEntry> _primaryCache = new Dictionary<string, CacheEntry>();
+        private readonly Dictionary<string, CacheEntry> _primaryCache = new Dictionary<string, CacheEntry>(StringComparer.OrdinalIgnoreCase);
```

Comment:
This was originally comparing ProjectNames. https://github.com/NuGet/NuGet.Client/commit/3d091dad054331cc212651ad6d88d18533d1915f#diff-5a34d358c305cf4b9fb65e2fba45c849406b2898a65aa0a75513e41370c2161aR19

[ProjectNames.Equals](https://github.com/NuGet/NuGet.Client/blob/0bf00d2496ec6de66fe67b1c8f571090ac2eef87/src/NuGet.Clients/NuGet.VisualStudio.Common/ProjectSystems/ProjectNames.cs#L243-L256) does a case insensitive check over the full name, so this change restores the dictionary to be case insensitive.